Reorganizing environment object schema docs (#7868)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-exposures.mdx

---
title: "Exposures object schema"
sidebar_label: "Exposures"
id: "discovery-schema-environment-applied-exposures"
---
 
import { NodeArgsTable, SchemaTable } from "./schema";

> [Exposures](/docs/build/exposures) are dbt resources that represent downstream uses of your project, such as dashboards, applications, or data science pipelines.You can query exposures through the Discovery API to understand which assets depend on your models.

Reviewer: ```suggestion
[Exposures](/docs/build/exposures) are dbt resources that represent downstream uses of your project, such as dashboards, applications, or data science pipelines. You can query exposures through the Discovery API to understand which assets depend on your models.
```

The [Example query](#example-query) illustrates a few fields you can query with the `exposures` object. Refer to [Fields](#fields) to view the entire schema, which provides all possible fields you can query.

### Arguments

When querying for `exposures`, you can use the following arguments:

> <NodeArgsTable parent="AppliedState" name="exposures" />

Reviewer: `filter` Type -- is it empty?

### Example query

> You can specify the `environmentId`, `first: 100`, and filter by model `uniqueIds` to return all the downstream exposures (dashboards, applications, etc.) that depend on the 'customers' model in the 'marketing' package, limited to the first 100 results:

Reviewer: ```suggestion
You can specify the `environmentId`, `first: 100`, and filter by model `uniqueIds` to return all the downstream exposures (dashboards, applications, etc.) that depend on the `customers` model in the `marketing` package, limited to the first 100 results:
```

```graphql
query {
  environment(id: 834) {
    applied {
      exposures(
        filter: {
          uniqueIds: ["model.marketing.customers"] # Use this format for unique ID: RESOURCE_TYPE.PACKAGE_NAME.RESOURCE_NAME
        }, 
        first: 100
      ) {
        edges {
          node {
            accountId
            exposureType
            fqn
            projectId
            url
          }
        }
      }
    }
  }
}
```

### Fields

When querying for `exposures`, you can use the following fields:

> <SchemaTable nodeName="ExposureAppliedStateNodeConnection" />

Reviewer: Fields have no description

> ### Key Fields from Nodes

Reviewer: ```suggestion
### Key fields from nodes
```

> <SchemaTable nodeName="ExposureAppliedStateNode" />

Reviewer: Description for `url` should be: Exposure URL
